nrf_wifi/fw_if/umac_if/src/tx.c

```diff
@@ -1108,24 +1347,76 @@ enum nrf_wifi_status tx_done_process(struct nrf_wifi_fmac_dev_ctx *fmac_dev_ctx,
 
 	def_dev_ctx->host_stats.total_tx_done_pkts += pkt;
 
-	pkts_pending = tx_buff_req_free(fmac_dev_ctx, config->tx_desc_num, &queue);
+	pkts_pending = tx_buff_req_free(fmac_dev_ctx, tx_desc_num, &queue);
 
 	if (pkts_pending) {
-		if (def_dev_ctx->twt_sleep_status ==
-		    NRF_WIFI_FMAC_TWT_STATE_AWAKE) {
+#ifdef CONFIG_NRF700X_RAW_DATA_TX
+		unsigned int magic_num;
+		unsigned char *data = NULL;
+		struct nrf_wifi_fmac_vif_ctx *vif_ctx;
+		unsigned char if_idx;
+
+		pkt_info = &def_dev_ctx->tx_config.pkt_info_p[desc];
+		txq = pkt_info->pkt;
+
+		/**
+		 * we need to peek into the pending buffer to determine if
+		 * packet is a raw packet or not
+		 */
+		nwb = nrf_wifi_utils_list_peek(fmac_dev_ctx->fpriv->opriv,
+					       txq);
 
-			pkt_info = &def_dev_ctx->tx_config.pkt_info_p[desc];
+		data = nrf_wifi_osal_nbuf_data_get(fmac_dev_ctx->fpriv->opriv,
+						   nwb);
 
-			txq = pkt_info->pkt;
+		nrf_wifi_osal_mem_cpy(fmac_dev_ctx->fpriv->opriv,
+				      &magic_num,
+				      data,
+				      sizeof(unsigned int));
 
-			status = tx_cmd_init(fmac_dev_ctx,
-					     txq,
-					     desc,
-					     pkt_info->peer_id);
+		if (magic_num != NRF_WIFI_MAGIC_NUM_RAWTX) {
```

Comment:
Why not just do `if (*(unsigned int *)data != NRF_WIFI_MAGIC_NUM_RAWTX) {` here and avoid a mem_cpy above?

Reply:
i believe it is cleaner this way. but okay

Reply:
done